### PR TITLE
Fix postgres cert renewal issue

### PR DESCRIPTION
**Pull request by fydai**
I tested this on lethe and puppet is finally working again.

### CODE REVIEW

File: modules/ocf_postgres/manifests/init.pp [hunk 1]
```diff
@@ -32,6 +32,22 @@
       };
   }
 
+  # Postgres expects the letsencrypt certs to be owned by root
+  # The renewal scrypt expects the letsencrypt certs to be owned by ocfletsencrypt
```
Comment (jvperrin) [inline modules/ocf_postgres/manifests/init.pp]: nit: `scrypt` -> `script`

File: modules/ocf_postgres/manifests/init.pp [hunk 2]
```diff
@@ -32,6 +32,22 @@
...
+  # We change the owner to ocflets-encrypt before the renewal script runs
```
Comment (jvperrin) [inline modules/ocf_postgres/manifests/init.pp]: nit `ocflets-encrypt` -> `ocfletsencrypt`

File: modules/ocf_postgres/manifests/init.pp [hunk 3]
```diff
@@ -32,6 +32,22 @@
...
+    before  => Class['Ocf::Ssl::Default'],
+  }
+
+  exec {'chown-root':
+    command => 'chown -R root:ssl-cert /var/lib/lets-encrypt/certs',
+    require => Class['Ocf::Ssl::Default'],
+  }
+
+  # Change owner of /var/lib/lets-encrypt-certs to ocfletsencrypt
+  # postgres expects Class
```
Comment (jvperrin) [inline modules/ocf_postgres/manifests/init.pp]: I think both of these comments are too far down? At least they don't appear to pertain to the line below here.

File: modules/ocf_postgres/manifests/init.pp [hunk 4]
```diff
@@ -32,6 +32,22 @@
...
+  # Postgres expects the letsencrypt certs to be owned by root
+  # The renewal scrypt expects the letsencrypt certs to be owned by ocfletsencrypt
+  # We change the owner to ocflets-encrypt before the renewal script runs
+  # Then we change it back after the renewal script runs
+  exec { 'chown-letsencrypt':
+    command => 'chown -R ocfletsencrypt:ssl-cert /var/lib/lets-encrypt/certs',
+    before  => Class['Ocf::Ssl::Default'],
+  }
+
+  exec {'chown-root':
```
Comment (jvperrin) [inline modules/ocf_postgres/manifests/init.pp]: nit: space between `{` and `'`

**Comment by fydai** [comment]
Having the owner as ocfletsencrypt causes postgres ssl to not work, as postgres expects the owner to be root. The only workaround I've found online is to copy over the letsencrypt certs somewhere else to be owned by root/postgres. (https://serverfault.com/questions/807024/use-letsencrypt-certificates-for-postgresql). Jamming in the execs was the best solution I came up with to allow 1) keep the certs owned by root when postgres needs to use it and 2) when renewing, have the certs owned by ocfletsencrypt

**Comment by abizer** [comment]
I rejected this approach when I was working on it earlier because it's a one off hack. The correct way to address this, IMO, is to amend the logic in `ocf::ssl` to set permissions correctly based on what the `require`ing class requests as the ultimate owners of the certs. Hence the addition of the `owner` and `group` parameters to the class back when I added this module.

**Comment by fydai** [comment]
I don't see a particularly elegant way of solving this issue in `ocf::ssl` either. How could it ensure that the final owner could be set as `root:ssl_cert`, but requiring dehydrated run as `ocfletsencrypt:ssl_cert` without just moving the `exec`s into `ocf::ssl`?. I think this is fine as a solution because it addresses the issue of postgres not getting its cert updated, and it is reasonably clear that it is a workaround for an issue for an interaction between postgres and dehydrated.

**Pull request merged**